src/parquet/column/reader.cc

```diff
@@ -169,7 +169,8 @@ bool TypedColumnReader<DType>::ReadNewPage() {
             throw ParquetException("Unknown encoding type.");
         }
       }
-      current_decoder_->SetData(num_buffered_values_, buffer, data_size);
+      current_decoder_->SetData(
+          num_buffered_values_, buffer, static_cast<int>(data_size));
```

Comment:
We should use a more precise integer type in `SetData`, and the decoders. We can leave this as is though